File: csharp_body_deref/Scripts/Settings/InGameCharacterSettings.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Settings.InGameCharacterSettings
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Settings/InGameCharacterSettings.c
// Prior real C# source (older reference): Assets/Scripts/Settings/InGame/InGameCharacterSettings.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace Settings;

[Token(Token = "0x2000089")]
internal class InGameCharacterSettings : BaseSettingsContainer
{
	[Token(Token = "0x40002FE")]
	[FieldOffset(Offset = "0x20")]
	public IntSetting ChooseStatus;

	[Token(Token = "0x40002FF")]
	[FieldOffset(Offset = "0x28")]
	public StringSetting CharacterType;

	[Token(Token = "0x4000300")]
	[FieldOffset(Offset = "0x30")]
	public StringSetting Loadout;

	[Token(Token = "0x4000301")]
	[FieldOffset(Offset = "0x38")]
	public StringSetting Special;

	[Token(Token = "0x4000302")]
	[FieldOffset(Offset = "0x40")]
	public IntSetting CustomSet;

	[Token(Token = "0x4000303")]
	[FieldOffset(Offset = "0x48")]
	public IntSetting Costume;

	[Token(Token = "0x4000304")]
	[FieldOffset(Offset = "0x50")]
	public StringSetting Team;

	[Token(Token = "0x6000329")]
	[Address(RVA = "0x3F8A990", Offset = "0x3F8A990", VA = "0x3F8A990")]
	public InGameCharacterSettings()
	{
	}
}

File: cpp/Settings/InGameCharacterSettings.c
// Type: Settings.InGameCharacterSettings
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Settings/InGameCharacterSettings.cs
// Prior real C# source (older reference): Assets/Scripts/Settings/InGame/InGameCharacterSettings.cs
// --------------------------------

// Settings.InGameCharacterSettings$$.ctor
// il2cpp: void Settings_InGameCharacterSettings___ctor (Settings_InGameCharacterSettings_o* __this, const MethodInfo* method);
// 0x408a990

void Settings_InGameCharacterSettings___ctor(Settings_InGameCharacterSettings_o *__this,MethodInfo *method)

{
  System_String_o *pSVar1;
  Settings_IntSetting_o *pSVar2;
  Settings_StringSetting_o *pSVar3;
  
  if (g_data_057ac3ee == '\0') {
    il2cpp_runtime_helper_023445d0(&TypeInfo_IntSetting);
    il2cpp_runtime_helper_023445d0(&TypeInfo_StringSetting);
    il2cpp_runtime_helper_023445d0(&"Blue");
    g_data_057ac3ee = '\x01';
  }
  pSVar2 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar2,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).ChooseStatus = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).ChooseStatus,pSVar2);
  pSVar1 = (System_String_o *)**(undefined8 **)(g_data_057b9c00 + 0xb8);
  pSVar3 = (Settings_StringSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_StringSetting);
  Settings_StringSetting___ctor_40f74f0(pSVar3,pSVar1,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).CharacterType = pSVar3;
  il2cpp_runtime_helper_022b4080(&(__this->fields).CharacterType,pSVar3);
  pSVar1 = (System_String_o *)**(undefined8 **)(g_data_057b9c00 + 0xb8);
  pSVar3 = (Settings_StringSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_StringSetting);
  Settings_StringSetting___ctor_40f74f0(pSVar3,pSVar1,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).Loadout = pSVar3;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Loadout,pSVar3);
  pSVar1 = (System_String_o *)**(undefined8 **)(g_data_057b9c00 + 0xb8);
  pSVar3 = (Settings_StringSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_StringSetting);
  Settings_StringSetting___ctor_40f74f0(pSVar3,pSVar1,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).Special = pSVar3;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Special);
  pSVar2 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar2,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).CustomSet = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).CustomSet);
  pSVar2 = (Settings_IntSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_IntSetting);
  Settings_IntSetting___ctor_40f39a0(pSVar2,0,-0x80000000,0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).Costume = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Costume,pSVar2);
  pSVar3 = (Settings_StringSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_StringSetting);
  Settings_StringSetting___ctor_40f74f0(pSVar3,"Blue",0x7fffffff,(MethodInfo *)0x0);
  (__this->fields).Team = pSVar3;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Team);
  Settings_BaseSettingsContainer___ctor((Settings_PresetSettingsContainer_o *)__this,(MethodInfo *)0x0);
  return;
}


